source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -0,0 +1,135 @@
+
+![](../../../../images/wiki/upload-external-ova.png)  
+
+The process differ from import of OVA generated by oVirt in the following steps:  
+6. Only one temporary disk is created on the storage domain.  
+10. The whole OVA is streamed into the created disk.  
+11. Additional step that converts the streamed OVA and removed the temporary disk.  
+
+Note:  
+
+1. This process if not efficient in time and space, due to the creation of the temporary disk. Thus, we will suggest to make the OVA accessible to the host instead, especially in the case of mass import of OVA files. However, this process can be handy for a one upload with loose time- and space- constraints.  
+2. Therefore, it is a second-priority task.
+
+## Exporting a VM or Template as an OVA
+The user will be able to specify a VM (or a template) to be exported as an OVA. In that case, the user needs to provide a path on a host where the OVA file should be created.  
+
+The following figure depicts the process of exporting a VM with a single disk as OVA:  
+
+![](../../../../images/wiki/export-ova.png)  
+
+1. The front-end sends the back-end a VM and a path on a host.
+2. The back-end asks imagio-proxy to initialize a file at the given path with an estimated size of the produced OVA.
+3. Imageio creates the OVA file.
+4. The back-end sends imageio-proxy the disk to stream, which sends it to imageio.
+5. Imageio reads the disk and streams it into a QCOW volume within the OVA.
+6. The back-end sends the complete OVF specification to imageio-proxy, which sends it to imageio that streams it into the OVA.
+
+Notes:
+
+1. Step 5 is equivalent to do qemu-img convert with the collapse option and the target format being QCOW, on-the-fly.
+2. The streaming in step 5 should be done in a certain offset so the streaming of the OVA would be done into the beginning of the file (the OVF should be the first tar entry inside the OVA according to the OVA specification).
+3. The back-end should receive the overall amount of data written to the disk entry in step 4 so it could be placed inside the OVF.  
+4. The OVA file will be created with a special extension (step 3) and would be renamed (that extension would be dropped) at the end of the process (step 6). That is done in order to enable tools that synchronize shares between various deployments differentiate partial files and complete files.  
+
+## An Enhancement for Import OVAs
+Importing an OVA that was generated by oVirt won't involve virt-v2v and will be similar to the upload-ova process:  
+
+![](../../../../images/wiki/import-ova.png)  
+
+1. The front-end sends the back-end a VM and a path on a host.  
+2. The back-end retrieved via an agent (VDSM/Imageio/Other) the OVF configuration from within the OVA.  
+3. The back-end parses the OVF configuration and returns a VM to the front-end.
+4. The front-end sends (an optionally customized) parameters of the import process to the back-end.
```

Comment:
Suggested edit:
The front-end sends (optionally customized) parameters of the import process to the back-end.